stdlib/public/core/ManagedBuffer.swift

```diff
@@ -34,22 +34,14 @@ open class ManagedBuffer<Header, Element> {
       ManagedBuffer<Header, Element>) throws -> Header
   ) rethrows -> ManagedBuffer<Header, Element> {
 
-    let p = try ManagedBufferPointer<Header, Element>(
-      bufferClass: self,
-      minimumCapacity: minimumCapacity,
-      makingHeaderWith: { buffer, _ in
-        try factory(
-          unsafeDowncast(buffer, to: ManagedBuffer<Header, Element>.self))
-      })
-
-    return unsafeDowncast(p.buffer, to: ManagedBuffer<Header, Element>.self)
-  }
+    let p = Builtin.allocWithTailElems_1(
+         self,
+         minimumCapacity._builtinWordValue, Element.self)
 
-  /// Destroy the stored Header.
-  deinit {
-    ManagedBufferPointer(self).withUnsafeMutablePointerToHeader {
-      _ = $0.deinitialize()
-    }
+    let initHeaderVal = try factory(p)
+    p.headerAddress.initialize(to: initHeaderVal)
+    _fixLifetime(p) 
```

Comment:
@eeckstein Subtle insights like this need to be commented. Otherwise the next code maintainer just thinks you were being paranoid. If the optimizer can move a release above a potentially aliasing UnsafePointer access, then why wouldn't withUnsafePointer() have the same problem? Or maybe you're concerned that UnsafePointer<Header> does not appear to alias with ManagedBuffer<Header, Element>? That's even more subtle!